proposals/evaluation/uc-3-inheritance.md

```diff
@@ -0,0 +1,84 @@
+| acl     | http://www.w3.org/ns/auth/acl#    |
+| acp     | http://www.w3.org/ns/solid/acp#   |
+| ex      | https://example.com/              |
+| ldp     | http://www.w3.org/ns/ldp#         |
+| vcard   | http://www.w3.org/2006/vcard/ns#  |
+
+## Read access on a collection of resources
+
+See also: https://solid.github.io/authorization-panel/authorization-ucr/#inheritance-readonly
+
+Alice can read all resources in a collection.
+
+### ACP
+
+```turtle
+ex:Collection1
+  acp:accessControl ex:AccessControl1 ;
+  ldp:contains ex:Resource1 .
+
+ex:AccessControl1
+  acp:applyMembers ex:Policy1 .
+
+ex:Policy1
+  acp:anyOf ex:AgentMatcher1 ;
+  acp:allow acl:Read .
+
+ex:AgentMatcher1
+  acp:agent ex:Alice .
+```
+
+Note: Resources linked to a collection via `ldp:contains` will inherit policies linked to the collection's access control via `acp:applyMembers`. Collections linked to a collection via `ldp:contains` will pass the inherited policies through to their member resources.
+
+### WAC
+
+The Weekly status collection is an `ldp:BasicContainer`, which contains a number of `ldp:BasicContainers`, one for each weekly meeting. The advantage of having these as containers rather than plain resources is that any number of other documents can be added to the container too.
+
+```turtle
+# Resource: </weekly-status/>
+<.>
+  a ldp:BasicContainer ;
+  ldp:contains <2021-04-28/>, <2021-05-05/>, <2021-05-12/> .
+```
+
+The `</weekly-status/.acl>` resource is advertised as `</weekly-status/>`'s access control list via a `Link` header with a relationship type of `http://www.w3.org/ns/auth/acl#accessControl`.
+
+We have the following hierarchy of resources:
+
+```
+</weekly-status/>
+</weekly-status/.acl>
+</weekly-status/2021-04-28/>
+</weekly-status/2021-05-05/>
+</weekly-status/2021-05-12/>
+```
+
+Bob and Alice are members of the research group `</groups/research#g1>`:
+
+```turtle
+# Resource: </groups/research>
+<#g1>
+    a                vcard:Group ;
+    vcard:hasMember  ex:Bob ;
+    vcard:hasMember  ex:Alice .
+```
+
+The acl enabling read access to all resources contained by `</weekly-status/>` for all members of group `</groups/research#g1>` is:
+
+```turtle
+# Resource: </weekly-status/.acl>
+[]
+  acl:agentGroup </groups/research#g1> ;
+  acl:default <.> ;
+  acl:mode acl:Read .
+```
+
+Note: A resource will use its parent container's access control `acl:default` authorization unless it has its own access control defined (parentage being defined via the `ldp:contains` predicate). Inheritance is not granular and can be cut off by adding an access control to a resource at any point in the resource hierarchy tree. If the parent container doesn't have an access control, the WAC inheritance algorithm works its way up through `ldp:contains` predicates.
+
+The downside of this is that adding an access control resource, requires all the relevant rules to be copied to the new acr, creating duplication of rules.
```

Comment:
Adding an "access control resource" to what? Why is it framed in terms of copying/duplicating rules as opposed to providing specific authorization policies for a given resource.

Reply:
We're just trying to compare features or characteristics of WAC and ACP.

Providing specific authorization policies is something both can do, policy reuse seems to be, to my understanding, something WAC currently doesn't allow.

Reply:
I think we need to extend the example with an action where the ACL of one of the weekly meetings is edited with a `PUT`. 
Say we want to add another person who can read those weekly meeting notes. (Can we find that explicitly referenced in the use cases, so that we can build on the use cases?
We will find then that WAC requires the old inherited rules to be added to the PUT.

Reply:
inheritance-readonly could've been written in a couple of sentences. Not sure why Requirement 3.4.1 is listing all the things that's excluded, as opposed to only saying what's required.

Can you point to the requirements of UC3 where "policy reuse" is anything above and beyond applying applicable rules from somewhere? Why is "policy reuse" or rather copying/duplication being "systematically" evaluated here?

Reply:
Re. "Policy reuse" is just a specific instance of [DRY: Don't Repeat Yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), software engineering principle.  It is a noticeable difference between current WAC/ACL and [WAC+imports](https://github.com/solid/authorization-panel/issues/210#issuecomment-838747077), which I want to add to the list here too.

Reply:
The underlying issue with the use case is that it is solutions engineered as per "inheritance". If it had been centred on "policy reuse", then there'd more potential solutions, and the "inherited", "imports", "copying/duplicating" etc can be among those. Then one can come up with an actual/useful evaluation. For instance, comparing the type of inheritance mechanism approach taken by different solutions, or comparing how one uses inheritance, the other uses dynamic importing.. performance, maintainability etc. None of this is apparent in this "systematic" evaluation. Not sure what the end goal is to other than saying WAC has downsides and ACP has none? Alright.. go for it.

Reply:
@csarven 
> Not sure what the end goal is to other than saying WAC has downsides and ACP has none? Alright.. go for it.

I don't find that type of comment very helpful, and it is quite clearly wrong: We are just starting a process of comparing different ways of doing things. I for one am not prejudging any answer to these problems. There are always pros and cons to everything. As it happens, I am advocating for a minimal extension to WAC, which I believe could also be adopted by ACP.  The exercise here is to pinpoint how this can work, what the advantages are, etc... 

As I mentioned I will add a section for WAC+`:imports` to the document.
I have just implemented all the infrastructure for that in [commits to Reactive-Solid since May 20](https://github.com/co-operating-systems/Reactive-SoLiD/commits/master) and hope to get the access control based on that implemented over the weekend.

Reply:
Pardon me.. it is just the age old baggage on "WAC limitations" appearing in different forms. So, what are the "downsides" of ACP here? Missing data or there are none?

Reply:
I need to study the RDF @matthieubosquet put forward. But I think in teleconference he was saying that ACP also had the data duplication problem:  (without `:imports`) it seems like a lot of triples need to be copied over. That is at least how @kjetilk interpreted the specs in [issue 206 on efficiency](https://github.com/solid/authorization-panel/issues/206). But I'll leave @matthieubosquet to reply. If that is a problem, it would be good to make it clear.

Reply:
Perhaps I've misinterpreted this "downsides" only for WAC since it came right after WAC's Note. Perhaps need to have another section with a different heading covering this information for all solutions.

Reply:
I wrote the downsides for WAC because I am pretty sure I understand WAC, but not so sure about ACP.  
Also the ACP section had not been formatted consistently so that it would match WAC section RDF layout. This is still work in progress. 

But yes, I think it would be good to have pros and cons for each, hopefully structured in the same way, so that the differences and similarities become clear.